MineSweeper.py

```diff
@@ -0,0 +1,261 @@
+import Tkinter as tk
+import random
+
+import sys
+sys.path.append('~/Library/Python/2.7/site-packages')
+
+from drawingpanel import *
+
+BOARD_HEIGHT = 300
+BOARD_WIDTH = 300
+BOX_SIZE = 20
+
+NUMBER_MINES = ((BOARD_WIDTH / BOX_SIZE - 1)**2)//10 
+
+class Game():
+
+  def __init__(self):
+    self._master = tk.Tk()
+    self._panel = Canvas(self._master, width = BOARD_WIDTH, height = BOARD_HEIGHT + 100, bg = 'white')
+    self._panel.pack()
+
+    self._board = Board(self._panel)
+
+    self._board.assign_mines()
+
+    self._master.bind("<Button-1>", self.left_click_event)
+    self._master.bind("<Button-2>", self.right_click_event)
+
+     
+  def game_over(self):
+
+    self._master.unbind("<Button-1>")
+    self._master.unbind("<Button-2>")
+
+    self._b = Button(self._master, text = 'Play Again?', command = self.new_game)
+    self._panel.create_text(BOARD_WIDTH/2, BOARD_HEIGHT + 40, text  = 'Game Over', fill = 'red') 
+    self._b.pack()
```

Comment:
So for the stuff above where you're doing `x.a.b()` and you're saying "X shouldn't know about B!"... that's not really true in the `self` context. So these are all okay, IMO. It's like "Game knows it has a thing called `_master`, and it knows how `_master` works". There's some examples I'll comment on below that aren't okay [=